### PR TITLE
Delay xrfi a priori flags

### PR DESCRIPTION
**Pull request by adampbeardsley**
This PR goes hand in hand with one on hera_opm. The goal is to run xrfi on calfits and model vis files before running the delay filter + xrfi on data.
Biggest changes here are updating the top level script to do what is described above, allowing xrfi_run to not take an input file, and a new argument parser to handle this.

### CODE REVIEW

File: hera_qm/utils.py [hunk 1]
```diff
@@ -132,6 +135,75 @@ def get_metrics_ArgumentParser(method_name):
                        'visibilities formed with these antennas will be set to True.')
         a.add_argument('filename', metavar='filename', nargs='*', type=str, default=[],
                        help='file for which to flag RFI (only one file allowed).')
+    elif method_name == 'delay_xrfi_run':
```
Comment (jsdillon) [inline hera_qm/utils.py]: Are you familiar with argument groups? e.g. https://github.com/HERA-Team/hera_cal/blob/7ee5af6186e278ef8cd627cc81b2853e3e62d88a/hera_cal/smooth_cal.py#L430

Might make this whole thing more intelligible.

Comment (adampbeardsley) [reply hera_qm/utils.py]: I saw you doing that in hera cal. It's just an organizational thing right? Doesn't actually change behavior?
Are you suggesting using it within one of these "methods", or to replace the whole if/else block?

File: hera_qm/xrfi.py [hunk 2]
```diff
@@ -265,7 +266,12 @@ def xrfi_run(indata, args, history):
     model visibility file, and run an RFI-flagging algorithm to identify contaminated
     observations. Each set of flagging will be stored, as well as compressed versions.
     """
-    if isinstance(indata, UVData):
+    if indata is None:
+        if (args.model_file is None) and (args.calfits_fils is None):
+            raise AssertionError('Must provide at least one of: filename, '
```
Comment (jsdillon) [inline hera_qm/xrfi.py]: filename --> indata?

File: hera_qm/utils.py (hunk 1 above)
Comment (jsdillon) [reply hera_qm/utils.py]: It's purely organizational. And it makes the -h much neater.

**Comment by adampbeardsley** [comment]
@jsdillon I added a test, which indeed caught a bug. Could you review again?

**Comment by jsdillon** [comment]
oooh, look at that fun typo `calfits_fils`

Looks good to me.

**Pull request merged**